syft/rekor/rekor_test.go

```diff
@@ -0,0 +1,223 @@
+	var uuids []string
+	for uuid := range a.test.uuidsToLogEntryFile {
+		uuids = append(uuids, uuid)
+	}
+
+	res := &index.SearchIndexOK{
+		Payload: uuids,
+	}
+	return res, nil
+}
+
+/*
+***************
+
+	Test function
+
+****************
+*/
+
+func Test_CreateRekorSbomRels(t *testing.T) {
+	default_tc := &http.Client{
+		Transport: roundTripperMock{sbomFile: "test-fixtures/sboms/sbom-1.txt"},
+	}
+
+	testLogger, hook := test.NewNullLogger()
+	log.Log = testLogger
+
+	// just test succesful calls here because CreateRekorSbomRels does not return the verification errors, parse errors, etc
+	tests := []testCase{
+		{
+			name:          "one sbom entry, one non-sbom entry",
+			inputFilePath: "test-fixtures/file-to-hash.txt",
+			expectedOutput: []artifact.Relationship{
+				{
+					From: source.NewLocation("test-fixtures/file-to-hash.txt"),
+					To: NewExternalRef(
+						"SBOM-SPDX-ba96f4cc-d9e3-4c83-a1db-ec5456b6a9ce",
+						"http://www.example.com/binary.spdx",
+						spdx.SHA1,
+						"eb141a8a026322e2ff6a1ec851af5268dfe59b20",
+					),
+					Type: artifact.ContainsRelationship,
+				},
+			},
+			uuidsToLogEntryFile: map[string]string{
+				"c71d239df91726fc519c6eb72d318ec65820627232b2f796219e87dcf35d0ab4": "test-fixtures/log-entries/log-entry-1.json",
+				"88aa67ce4f4a3fa3e8da8adb4e4799b53372f078459639e571e5583e2685c304": "test-fixtures/log-entries/log-entry-2.json",
+			},
+		},
+		{
+			name:          "sbom missing namespace",
+			inputFilePath: "test-fixtures/file-to-hash.txt",
+			uuidsToLogEntryFile: map[string]string{
+				"09f4d6138d167fc246dc69badb11b9a931395e7ca00fb38a1889d287f9d4110e": "test-fixtures/log-entries/log-entry-5.json",
+			},
+			httpClient: &http.Client{
+				Transport: roundTripperMock{sbomFile: "test-fixtures/sboms/sbom-3.txt"},
+			},
+			expectedLog: "SBOM found on Rekor for file in location test-fixtures/file-to-hash.txt, but its namespace is empty. Ignoring SBOM.",
+		},
+		{
+			name:          "file to hash is a folder",
+			inputFilePath: "test-fixtures/",
+			expectedErr:   "error generating digests",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// set up mocks
+			rekorMock := &client.Rekor{
+				Entries: entriesMock{test: test},
+				Index:   indexMock{test: test},
+			}
+
+			var httpClient *http.Client
+			if test.httpClient == nil {
+				httpClient = default_tc
+			} else {
+				httpClient = test.httpClient
+			}
+
+			client := &Client{
+				rekorClient: rekorMock,
+				httpClient:  httpClient,
+			}
+
+			resolver := source.NewMockResolverForPaths(test.inputFilePath)
+			location := source.NewLocation(test.inputFilePath)
+
+			rels, err := CreateRekorSbomRels(resolver, location, client)
+
+			if test.expectedErr == "" {
+				assert.Equal(t, test.expectedOutput, rels)
+				assert.NoError(t, err)
+			}
+			if test.expectedLog != "" {
+				if hook.LastEntry().Message != test.expectedLog {
```

Comment:
if the behavior for `expectedLog` only the last entry? or existence within all log messages? will adding a new log statement break the test?